### PR TITLE
chore(stories): add categories

### PR DESCRIPTION
**Pull request by francoischalifour**
## Description

This adds sections in our Storybook following the same organization as the [documentation API ref](https://www.algolia.com/doc/api-reference/widgets/js/).

Note that this **updates the stories URLs** and therefore we need to update the documentation links for v4.

Let me know if you want to go that route so that I PR the doc repo.

## Preview

[See stories →](https://deploy-preview-4154--instantsearchjs.netlify.com/stories)

### CODE REVIEW

File: stories/instantsearch.stories.js [hunk 1]
```diff
@@ -27,4 +27,20 @@ storiesOf('InstantSearch', module)
         },
       },
     })
+  )
+  .add(
+    'with refresh to reload',
```
Comment (francoischalifour) [inline stories/instantsearch.stories.js]: This story used to be on its own but it makes more sense to have it in `instantsearch`.

**Comment by francoischalifour** [comment]
@samouss Do you have an easy solution in mind? Storybook's documentation isn't clear about sorting categories of stories.

**Comment by Haroenv** [comment]
already upvoted this once apparently but: https://github.com/storybookjs/storybook/issues/548#issuecomment-417202498

**Comment by Haroenv** [comment]
there's also https://github.com/storybookjs/storybook/issues/548#issuecomment-530305279

**Comment by francoischalifour** [comment]
Thanks @Haroenv! See 2d03a572ea140451010ba862337c931d1c30f448.

**Comment by francoischalifour** [comment]
Typing issues on Storybook's end... Let's wait for storybookjs/storybook#8233 to be fixed.

File: .storybook/config.ts [hunk 2]
```diff
@@ -1,13 +1,37 @@
...
     }),
+    storySort(a: any[], b: any[]) {
+      const categoryA = a[1].kind.split('|')[0];
+      const categoryB = b[1].kind.split('|')[0];
+
+      if (categories.indexOf(categoryA) === categories.indexOf(categoryB)) {
+        return 0;
+      }
+
+      return categories.indexOf(categoryA) < categories.indexOf(categoryB)
```
Comment (Haroenv) [inline .storybook/config.ts]:
```suggestion
      return categories.indexOf(categoryA) - categories.indexOf(categoryB)
```

this is the same no?

Comment (Haroenv) [reply .storybook/config.ts]: (or the other order, i'm always confused about that haha)

**Comment by Haroenv** [comment]
https://github.com/storybookjs/storybook/issues/7855

**Comment by Haroenv** [comment]
we can add `@types/webpack-env` as a dev dep, or skipLibCheck: true

**Comment by francoischalifour** [comment]
Upgrading to Storybook 5.2.3 fixed the issue.

**Comment by francoischalifour** [comment]
@Haroenv PR sent in algolia/doc#3807. I'm also updating the PR for the new `index` widget.

**Comment by francoischalifour** [comment]
The order of the sections changed [in the doc](https://www.algolia.com/doc/api-reference/widgets/js/) in the meantime ahah. I'll reflect the order in this PR.

**Pull request merged**